sdk/servicebus/azure-servicebus/tests/async_tests/test_queues_async.py

```diff
@@ -2074,6 +2074,77 @@ async def _hack_amqp_sender_run_async(self, **kwargs):
+                    await self._connection.sleep_async(self._backoff)
+                    self._backoff = 0
+                await self._connection.work_async()
+            else:
+                try:
+                    await self._link.update_pending_deliveries()
+                    await self._connection.listen(wait=self._socket_timeout, **kwargs)
+                except ValueError:
+                    self._shutdown = True
+                    return False
+            return True
+
+        async with ServiceBusClient.from_connection_string(
+                servicebus_namespace_connection_string, logging_enable=False, uamqp_transport=uamqp_transport) as sb_client:
+            async with sb_client.get_queue_sender(servicebus_queue.name) as sender:
+                # this one doesn't need to reset the method, as it's hacking the method on the instance
+                sender._handler._client_run_async = types.MethodType(_hack_amqp_sender_run_async, sender._handler)
+                with pytest.raises(OperationTimeoutError):
+                    await sender.send_messages(ServiceBusMessage("body"), timeout=5)
+
+        if not uamqp_transport:
+            # Amqp
+            async with ServiceBusClient.from_connection_string(
+                servicebus_namespace_connection_string,
+                uamqp_transport=uamqp_transport
+            ) as sb_client:
+                async with sb_client.get_queue_sender(servicebus_queue.name, socket_timeout=1.0) as sender:
+                    payload = "A" * 250 * 1024
+                    await sender.send_messages(ServiceBusMessage(payload))
+
+            if uamqp:
+                transport_type = uamqp.constants.TransportType.AmqpOverWebsocket
+            else:
+                transport_type = TransportType.AmqpOverWebsocket
+            # AmqpOverWebsocket
+            async with ServiceBusClient.from_connection_string(
+                servicebus_namespace_connection_string,
+                transport_type=transport_type,
+                uamqp_transport=uamqp_transport
+            ) as sb_client:
+                async with sb_client.get_queue_sender(servicebus_queue.name, socket_timeout=1.2) as sender:
+                    payload = "A" * 250 * 1024
+                    await sender.send_messages(ServiceBusMessage(payload))
+
+        # ReceiveMessages
+        async with ServiceBusClient.from_connection_string(
+                servicebus_namespace_connection_string, logging_enable=False, uamqp_transport=uamqp_transport) as sb_client:
+            async with sb_client.get_queue_receiver(servicebus_queue.name, max_wait_time=5) as receiver:
+                messages = await receiver.receive_messages(max_message_count=5, max_wait_time=5)
+                for message in messages:
+                    if not uamqp_transport:
+                        assert message._delivery_id is not None
```

Comment:
We should also assert here that the recvd message is the same as the sent message to check if payload stitching is working.

Reply:
yup yup!